### ISSUE TITLE
Client initiates data sync using the sync button

### ISSUE THREAD

**Issue by ihassin** [clintonhealthaccess/chailmis-android#255]
Scenario: Chaining with absolute numbers manually
  Given the server has "5" items of type "A"
  When I update SOH with "Received" "A" "10"
  And I update SOH with "Dispensed" "A" "5"
  And the manual sync succeeds
  Then the client updates SOH for "A" with value "5"
  And the server shows that there are "10" items of type "A" for my clinic

Client should examine return codes 200/201, 401, 500 to determine if the operation succeeded  or not.
Upon failure, the transaction should be cleared. The user will supply a new translation with  a new absolute number when communications have been re-established.

For example: SOH is 10, but the client cannot reach the server.
Tablet forgets this transaction after failing to contact the server for three times for example.

Since then, another 20 items were added to inventory, and the operator must try to sync with 30 this time.

**Comment by karlwbrown** [comment]
Itamar, I'm not sure about this story. It seems the client should send a SOH value of "10", not "5" in this case. 
re: return codes, DHIS2 has a rest api, so the success/failure may be more nuanced than return code 200, but I'm not positive, we'd have to ask Gary or HISP how failure/success is communicated via the rest api.

**Comment by garymabin** [comment]
I think whether the client should send "5" or "10" is determined by the
initial value, if the SOH sync works, if the server has 5 items, then the
client's initial value before the sync is "5", then "10" will be send to
server at that day.


Bin MaConsultantEmailbinma@thoughtworks.comTelephone+86 13125033274
<+86+13125033274>[image: ThoughtWorks]
<http://www.thoughtworks.com/?utm_campaign=bin-ma-signature&utm_medium=email&utm_source=thoughtworks-email-signature-generator>

2015-10-09 1:43 GMT+08:00 Karl Brown <notifications@github.com>:

**Comment by ihassin** [comment]
This is ready for dev, but should be picked up till #253 is at least code-complete

**Comment by garymabin** [comment]
Finish dev with #253

**Comment by ihassin** [comment]
Lovely!

- Itamar

**Comment by chai-lmis** [comment]
Tested: this works as expected, however the "stock Balance" on the Bin card (on the app) does not update to the new SOH value

**Comment by garymabin** [comment]
Still can not get what is " 'stock Balance' on the Bin card"，Is it another commodity with individual SOH or something else?

**Comment by ihassin** [comment]
@egwuma - please be very specific when describing bugs: Please list the state of the data when you ran the app, and the steps you took and list the expected values compared to the actual values.

**Comment by chai-lmis** [comment]
On the Bin card,  where you have "stock balance" near the Max/Min Values, the value does not match the SOH. Stock Balance = SOH at all times

**Comment by ihassin** [comment]
@garymabin will check this today

**Comment by garymabin** [comment]
Yes, I've figured out why the Stock Balance is not equal to SOH now, the SOH on the app was just for one day previously, so the Stock Balance at that time is calculated by adding SOH each day in a month. After we change the SOH  to always carrying on the previous day's value, the Stock Balance should just equal to the latest day's SOH rather than do the add up again.   Fixed fc71dbc

**Comment by chai-lmis** [comment]
same as card 253

**Comment by ihassin** [comment]
@egwuma - it seems the app side is ok as it's not giving any errors when syncing. I need a way to debug on the server side. Either direct access to its logs, or work with one of the devs on the server side to see transactions moving.

**Comment by ihassin** [comment]
I had Albendazole SOH = 10; I dispensed 5, I see SOH at 5 now. Bin shows 5 also. We'll wait another day, and dispense some more and expect to see correct results.

**Comment by ihassin** [comment]
Correct number appeared after sync at 4am. Closing this ticket.
